src/components/MainForm/Mainform.tsx

```diff
@@ -40,179 +135,184 @@ interface IFormInput {
 
 export default function Mainform() {
   const [open, setOpen] = React.useState(false);
-  // const [age, setAge] = React.useState<number | string>('');
+  const history = useHistory();
+  const [country, setCountry] = React.useState<any>(countries.USA);
 
-  // const handleChange = (event: SelectChangeEvent<typeof age>) => {
-  //   setAge(Number(event.target.value) || '');
-  // };
-
-  const handleClickOpen = () => {
-    setOpen(true);
+  const FlagsChanger = (country: any) => {
+    setCountry(country);
+    setOpen(false);
   };
 
-  const handleClose = (
-    event: React.SyntheticEvent<unknown>,
-    reason?: string
-  ) => {
-    if (reason !== "backdropClick") {
-      setOpen(false);
-    }
-  };
-  let pakFlag = "🇵🇰";
+  const formClasses = useStyles();
 
   const {
     register,
     handleSubmit,
+    setValue,
     formState: { errors },
   } = useForm<IFormInput>();
 
   const FormSubmitHandler: SubmitHandler<IFormInput> = (data: IFormInput) => {
+    history.push("./loading");
     console.log(data);
+
+    store.dispatch({
+      type: "ADD_DATA",
+      payload: data,
+    });
```

Comment:
Make a directory name Actions and make .ts file in it and put actions in that file